src/main/java/de/adorsys/keycloak/config/util/ProtocolMapperUtil.java

```diff
@@ -35,23 +33,19 @@ public static List<ProtocolMapperRepresentation> estimateProtocolMappersToRemove
             List<ProtocolMapperRepresentation> protocolMappers,
             List<ProtocolMapperRepresentation> existingProtocolMappers
     ) {
-        List<ProtocolMapperRepresentation> protocolMappersToRemove = new ArrayList<>();
-
-        if (existingProtocolMappers == null) {
-            return protocolMappersToRemove;
+        if (existingProtocolMappers == null || existingProtocolMappers.isEmpty()) {
+            return List.of(); // Return an immutable empty list
```

Comment:
Hi @Motouom 

The setting of `existingProtocolMappers.isEmpty()` must be done before reimporting. This requirement introduces a limitation. It could lead to potential errors if overlooked.  Consider the impact on workflow and efficiency.  A clear process is essential to avoid setbacks.  

![image](https://github.com/user-attachments/assets/a2e1c7f6-4687-44b5-ab73-03238a9fceda)

Reply:
Hello @AssahBismarkabah,

It's essential to understand the correct method for managing protocol mappers to ensure they are effectively deleted. If you remove the entire `protocolMappers` section from your configuration, any already imported mappers will remain intact.

To properly delete protocol mappers, follow these steps:

1. **To remove a specific mapper**, you must delete that specific entry from the `protocolMappers` array.
2. **To delete all protocol mappers**, you must leave the `protocolMappers` array in the configuration but ensure it is empty.

Here's how the process works:

Your initial configuration will look like this:

```json
{
    "realm": "test-realm",
    "enabled": true,
    "clients": [
        {
            "clientId": "test-client",
            "enabled": true,
            "clientAuthenticatorType": "client-secret",
            "redirectUris": ["*"],
            "webOrigins": ["*"],
            "protocolMappers": [
                {
                    "name": "TestMapper",
                    "protocol": "openid-connect",
                    "protocolMapper": "oidc-usermodel-attribute-mapper",
                    "consentRequired": false,
                    "config": {
                        "userinfo.token.claim": "true",
                        "user.attribute": "testAttribute",
                        "id.token.claim": "true",
                        "access.token.claim": "true",
                        "claim.name": "testClaim",
                        "jsonType.label": "String"
                    }
                }
            ]
        }
    ]
}
```

If you attempt to remove the `protocolMappers` section entirely, the configuration will look like this:

```json
{
    "realm": "test-realm",
    "enabled": true,
    "clients": [
        {
            "clientId": "test-client",
            "enabled": true,
            "clientAuthenticatorType": "client-secret",
            "redirectUris": ["*"],
            "webOrigins": ["*"]
        }
    ]
}
```

This action will not delete the protocol mapper, so do not take this approach.

To effectively ensure the removal of protocol mappers, adjust your configuration to this format:

```json
{
    "realm": "test-realm",
    "enabled": true,
    "clients": [
        {
            "clientId": "test-client",
            "enabled": true,
            "clientAuthenticatorType": "client-secret",
            "redirectUris": ["*"],
            "webOrigins": ["*"],
            "protocolMappers": [
                // This array must be left empty to indicate removal
            ]
        }
    ]
}
```

By maintaining the `protocolMappers` array while leaving it empty, you clearly communicate that there are no remaining protocol mappers.